src/charm.py

```diff
@@ -0,0 +1,309 @@
+            self, relation_name="default-database", database_name=DEFAULT_DATABASE_NAME
+        )
+
+        self.framework.observe(
+            self.on.default_database_relation_joined,
+            self._configure_amf,
+        )
+        self.framework.observe(
+            self.on.amf_database_relation_joined,
+            self._configure_amf,
+        )
+        self.framework.observe(self._default_database.on.database_created, self._configure_amf)
+        self.framework.observe(self._amf_database.on.database_created, self._configure_amf)
+        self.framework.observe(self.on.amf_pebble_ready, self._configure_amf)
+        self.framework.observe(self._nrf_requires.on.nrf_available, self._configure_amf)
+
+    def _configure_amf(
+        self,
+        event: EventBase,
+    ) -> None:
+        """Handle pebble ready event for AMF container.
+
+        Args:
+            event (PebbleReadyEvent, DatabaseCreatedEvent, NRFAvailableEvent): Juju event
+        """
+        if not self._amf_container.can_connect():
+            self.unit.status = MaintenanceStatus("Waiting for service to start")
+            event.defer()
+            return
+        for relation in ["fiveg_nrf", "amf-database", "default-database"]:
+            if not self._relation_created(relation):
+                self.unit.status = BlockedStatus(f"Waiting for {relation} relation")
+                return
+        if not self._default_database_is_available():
+            self.unit.status = WaitingStatus("Waiting for the default database to be available")
+            event.defer()
+            return
+        if not self._amf_database_is_available():
+            self.unit.status = WaitingStatus("Waiting for the amf database to be available")
+            event.defer()
+            return
+        if not self._get_default_database_info():
+            self.unit.status = WaitingStatus("Waiting for default database info to be available")
+            event.defer()
+            return
+        if not self._nrf_requires.nrf_url:
+            self.unit.status = WaitingStatus("Waiting for NRF data to be available")
+            event.defer()
+            return
+        if not self._amf_container.exists(path=CONFIG_DIR_PATH):
+            self.unit.status = WaitingStatus("Waiting for storage to be attached")
+            event.defer()
+            return
+        content = self._render_config_file(
+            ngapp_port=NGAPP_PORT,
+            sctp_grpc_port=SCTP_GRPC_PORT,
+            sbi_port=SBI_PORT,
+            nrf_url=self._nrf_requires.nrf_url,
+            amf_url=self._amf_hostname,
+            default_database_name=DEFAULT_DATABASE_NAME,
+            amf_database_name=AMF_DATABASE_NAME,
+            database_url=self._get_default_database_info()["uris"].split(",")[0],
+            full_network_name=CORE_NETWORK_FULL_NAME,
+            short_network_name=CORE_NETWORK_SHORT_NAME,
+        )
+        if not self._config_file_content_matches(content=content):
```

Comment:
I am not sure what this check does for us, we could just write out the file anyway and would get the same result.